src/core/qgsrendercontext.cpp

```diff
@@ -229,6 +239,12 @@ double QgsRenderContext::convertToPainterUnits( double size, QgsUnitTypes::Rende
       conversionFactor = mScaleFactor * INCH_TO_MM;
       break;
 
+    case QgsUnitTypes::RenderMetersInMapUnits:
+    {
+      size = convertMetersToMapUnits( size );
+      unit = QgsUnitTypes::RenderMapUnits;
+      // Fall through to RenderMapUnits with size in meters converted to size in MapUnits
```

Comment:
You'll need to put

    FALLTHROUGH;

here to flag that the fallthrough is intentional and to avoid the build warning